domains/transfer-agreements/TransferAgreements.API/API/TransferAgreementsAutomation/TransferAgreementsAutomationService.cs

```diff
@@ -32,16 +32,23 @@ public async Task Run(CancellationToken stoppingToken)
     {
         while (!stoppingToken.IsCancellationRequested)
         {
+
             logger.LogInformation("TransferAgreementsAutomationService running at: {time}", DateTimeOffset.Now);
-            metrics.ResetCertificatesTransferred();
-            var transferAgreements = await transferAgreementRepository.GetAllTransferAgreements();
 
-            foreach (var transferAgreement in transferAgreements)
+            try
             {
-                await projectOriginWalletService.TransferCertificates(transferAgreement);
-            }
+                var transferAgreements = await transferAgreementRepository.GetAllTransferAgreements();
+                metrics.SetNumberOfTransferAgreements(transferAgreements.Count);
 
-            metrics.SetNumberOfTransferAgreementsOnLastRun(transferAgreements.Count);
+                foreach (var transferAgreement in transferAgreements)
+                {
+                    await projectOriginWalletService.TransferCertificates(transferAgreement);
+                }
+            }
+            catch (Exception e)
+            {
+                logger.LogInformation("Something went wrong with the TransferAgreementsAutomationService: {exception}", e.Message);
```

Comment:
This should be a Warning and I think you should log the entire exception and not only the message:

```suggestion
                logger.LogWarning("Something went wrong with the TransferAgreementsAutomationService: {exception}", e);
```